odxtools/parameters/parameterbase.py

```diff
@@ -131,33 +132,37 @@ def encode_into_pdu(self, encode_state: EncodeState) -> bytes:
             the message's blob after adding the encoded parameter into it
 
         """
-        byte_value = self.get_coded_value_as_bytes(encode_state)
-
-        old_rpc = encode_state.coded_message
-        new_rpc = bytearray(old_rpc)
+        msg_blob = encode_state.coded_message
+        param_blob = self.get_coded_value_as_bytes(encode_state)
 
         if self.byte_position is not None:
             byte_position = self.byte_position
         else:
-            byte_position = len(old_rpc)
-
-        min_length = byte_position + len(byte_value)
-        if len(old_rpc) < min_length:
-            # Make byte code longer if necessary
-            new_rpc += bytearray([0] * (min_length - len(old_rpc)))
-        for byte_idx_val, byte_idx_rpc in enumerate(
-                range(byte_position, byte_position + len(byte_value))):
+            byte_position = len(msg_blob)
+
+        return self._insert_into_blob(msg_blob, param_blob, byte_position)
+
+    def _insert_into_blob(self, blob: bytes, new_data: bytes, pos: Optional[int] = None) -> bytes:
```

Comment:
this is not really an "insert", input is not modified

Reply:
okay, I changed it to `encode`...